### ISSUE TITLE
TypeError: Right-hand side of 'instanceof' is not an object

### ISSUE THREAD

**Issue by websy85** [angular/angular#10636]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Using the router (3.0.0-rc.1) with JavaScript and a simple route configuration -

```javascript
var routeConfig = [
  {
    path: '',
    redirectTo: 'home',
    pathMatch: 'full'
  },
  {
    path: 'home',
    component: HomeComponent
  },
  {
    path: 'other',
    component: OtherComponent
  }
];
```
With the routerLinks configured like this -

```html
<div>
  <a [routerLink]="['/']">Home</a>
  <a [routerLink]="['/other']">Other</a>
</div>
<router-outlet></router-outlet>
```

When clicking on a link the following error is produced - 

TypeError: Right-hand side of 'instanceof' is not an object

**Expected/desired behavior**

Should update the router-outlet with the correct component

I've resolved the problem by changing `Line 668` in `router.umd.js` from
```javascript
if (e instanceof rxjs_util_EmptyError.EmptyError) {
```
Where ```rxjs_util_EmptyError.EmptyError``` returns undefined, to 
```javascript
if (e instanceof rxjs_util_EmptyError) {
```
Rather than submit a pull request I thought it better to raise it as a bug. I'm not sure how the change impacts other parts of the router or if it would affect for example a Typescript implementation of it.

**Please tell us about your environment:**

* **Angular version:** 2.0.0-rc.X
 2.0.0-rc.5

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ] 
Chrome Version 52.0.2743.116 (64-bit)
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5] 
ES5

**Comment by forestofrain** [comment]
@websy85 thank you for the workaround.

**Comment by teak** [comment]
Can confirm same issue. No particular cause in my code that I can find, very basic routing setup. Angular 2 RC5, router 3.0.0-rc.1, using ES5. Workaround fixed issue (thanks a bunch for that).

**Comment by vsavkin** [comment]
This issue has been fixed in router 3.0.0.
